sqlglot/expressions.py

```diff
@@ -2998,6 +2998,30 @@ class DatetimeTrunc(Func, TimeUnit):
     arg_types = {"this": True, "unit": True, "zone": False}
 
 
+class DateOfWeek(Func, TimeUnit):
```

Comment:
why do these subclass TimeUnit if they don't take Unit, they should just be Func and pass